components/table/Sidebar.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from 'react'
+
+import tableClassNames from './index.css'
+
+import { CloseIcon } from 'design'
+
+class Sidebar extends React.Component {
+  constructor(props) {
+    super(props)
+    const { children } = props
+    if (children) React.Children.only(children)
+  }
+
+  render() {
+    const { isOpen, children, ...passProps } = this.props
+    return (
+      <div className={tableClassNames.sidebar}>
+        {children && React.cloneElement(children, passProps)}
```

Comment:
It would work only if the children is a single element. If it's more than one element, it will probably throw or pass props in bulk. You have no validation so I am not sure it's what you want to.

Reply:
How so? Validation is in the constructor. I know if you change sidebar during runtime it will probably fail but for now, I consider as an edge case and will fix it when we actually face this problem.

Reply:
Sorry, I missed the validation in the sidebar. I did not read the code carefully enough.